homeassistant/components/gardena_bluetooth/sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Support for switch entities."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta, timezone
+
+from gardena_bluetooth.const import Battery, DeviceConfiguration, Valve
+from gardena_bluetooth.parse import Characteristic
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import Coordinator, GardenaBluetoothEntity
+
+
+@dataclass
+class GardenaBluetoothSensorEntityDescription(SensorEntityDescription):
+    """Description of entity."""
+
+    char: Characteristic = field(default_factory=lambda: Characteristic(""))
+
+
+DESCRIPTIONS = (
+    GardenaBluetoothSensorEntityDescription(
+        key=Valve.activation_reason.uuid,
+        translation_key="activation_reason",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        char=Valve.activation_reason,
+    ),
+    GardenaBluetoothSensorEntityDescription(
+        key=Battery.battery_level.uuid,
+        translation_key="battery_level",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=PERCENTAGE,
+        char=Battery.battery_level,
+    ),
+    GardenaBluetoothSensorEntityDescription(
+        key=DeviceConfiguration.unix_timestamp.uuid,
```

Comment:
What is this timestamp for? It's name is just time, but what does it actually do?

Reply:
It's pretty much what time the device thinks the world is in. It generally not that useful that is why it's disabled by default. Maybe i should remove the entity.

Reply:
I think showing this information in system health or diagnostics would be a better place, as it doesn't make a lot of sense for people to make automations or get info from imo

Reply:
Agree, removed.